src/_igraph/convert.c

```diff
@@ -2684,6 +2684,73 @@ int igraphmodule_PyObject_to_vector_int_ptr_t(PyObject* list, igraph_vector_ptr_
   return 0;
 }
 
+/**
+ * \ingroup python_interface_conversion
+ * \brief Converts a Python list of lists to an \c igraph_vector_list_t
+ *        (containing \c igraph_vector_int_t items).
+ *
+ * \param o the Python object representing the list of lists
+ * \param m the address of an uninitialized \c igraph_vector_list_t
+ * \return 0 if everything was OK, 1 otherwise. Sets appropriate exceptions.
+ */
+int igraphmodule_PyObject_to_vector_list_t(PyObject* list, igraph_vector_list_t* vec) {
+  PyObject *it, *item;
+  igraph_vector_t *subvec;
+
+  if (PyUnicode_Check(list)) {
+    PyErr_SetString(PyExc_TypeError, "expected iterable (but not string)");
+    return 1;
+  }
+
+  it = PyObject_GetIter(list);
+  if (!it) {
+    return 1;
+  }
+
+  if (igraph_vector_list_init(vec, 0)) {
+    igraphmodule_handle_igraph_error();
+    Py_DECREF(it);
+    return 1;
+  }
+
+  /* TODO: what about this? is memory managed already? */
+  //IGRAPH_VECTOR_PTR_SET_ITEM_DESTRUCTOR(vec, igraph_vector_int_destroy);
+  while ((item = PyIter_Next(it)) != 0) {
+    /* FIXME: is this even the right way to do this? */
```

Comment:
No, this is not right.

You are using `push_back_new()` below. That function creates a new vector from scratch, adds it to the list, and returns a pointer to it. You cannot pass in your own vector. 

What you probably want is `push_back()`, which transfers the ownership of the vector. This means that you should _not_ destroy `subvec` once it's been inserted into the list. You should not touch it at all—it's gone, it lives in the list now. To access it, index into the list.

You do not need to allocate `subvec` on the heap. Keep things simple, use a local variable.

(But do not name these vec and subvec if possible, name them list and vec)

Reply:
yes, transfer ownership is what we need here, thank you for the hint

Just wording, but remember I didn't "name" any of those variables, the names were already there. I agree we should try and change names to something less confusing as we convert the whole codebase. Same for "keeping things simple":  I'm not coming up with these algorithms, they are already there, so I cannot "keep" things simple, but only *simplify* compared to what it is now. In any case, happy to simplify whenever possible, of course.

Reply:
Sorry, I didn't see the preceding function that you were basing this on. Yes, that was a "pointer vector" containing "vectors" so "vec of vecs" was consistent naming.  It also necessitated first allocating a vector_t dynamically before inserting it into the pointer vector.  Now we don't have to do that anymore.

Let me know if you have questions about `vector_list_t` until Tamás comes back. I don't know the Python API but I do know `vector_list_t` and how to move from a `vector_ptr_t`.

Reply:
Thanks @szhorvat ! These comments are already quite helpful.

I'm now in the thick of the woods with converting `int -> igraph_error_t` in attribute handling functions, such a nightmare. I'm sure I'll ask a few things about the new types lists, thanks